frontend/src/components/AboutUs/_aboutUs.scss

```diff
@@ -1,58 +1,161 @@
 @import '../../assets/variables';
 
+@mixin elementCard($font-bold, $background-color, $font-size: 20px) {
+  width: 100%;
+  height: 5.1rem;
+  text-transform: none;
+  font-size: $font-size;
+  font-weight: $font-bold;
+  font-family: "Alexandria", sans-serif;
+  border-radius: 1rem;
+  background-color: $background-color;
+
+  @media screen and ($smartphone-and-tablet-size) {
+    font-size: 12px;
+    height: 3.875rem;
+  }
+}
+
 .about-us {
   background-color: #f2dfd7;
   background-image: url("/images/bonebg.png");
+  background-repeat: none;
+  background-size: auto;
 
   &__container {
     max-width: 1140px;
     margin: 0 auto;
+    padding-bottom: 11.75rem;
 
     &__imgs {
       display: flex;
       justify-content: center;
       margin-right: 7.5rem;
 
+      @media screen and ($smartphone-and-tablet-size) {
+        margin-right: 2.5rem;
+      }
+
       img {
         height: 31.25rem;
+
+        @media screen and ($smartphone-and-tablet-size) {
+          height: 9rem;
+          width: calc(9rem + 3.9%);
```

Comment:
Qual é a linha pensamento por trás de todas essas constantes? Valores constantes em si não são um problema, o problema é o excesso deles e a inconsistência visual na hora de se ler o código. Não há comentários nem símbolos que indicam o porquê de `calc(9rem + 3.9%)`.

Recomendo que use variáveis e constantes simples; arredonde valores. `31.25rem` vira `500px`. Quem sabe uma variável comum `$offset` onde você usa `calc($offset / 2)` ou `calc($offset * 0.5)`, por exemplo. Mais simples e legível. Mesmo que não fique *exatamente* como o design proposto, mas próximo o suficiente e fiel ao conceito. Acho porcentagens com precisão nas casas decimais um tanto exagerado.

Evite usar `px` se possível, use `rem` para fontes, e evite alinhamentos com valores negativos. Essas coisas vão tornar o código mais fácil de se manter.

Reply:
E ae, Marcos
O que você quer dizer como constante? Se diz sobre os '@media', eles são para estilizar telas com proporções diferentes, se usei foi por que foi preciso alterar, caso tenha alguma dica ou ideia de como refatorar, é bem vindo! Me mande mensagem e conversamos.

Sobre o 'calc', coloquei pois cada imagem tem um tamanho diferente, não adiantaria eu definir um width padrão para cada, então colocando o 'calc', ele faz isso de forma dinâmica para mim. Os valores são: calc(o tamanho do height + a % que é maior no width). Coloquei a porcentagem definida especificamente, pois a width da imagem é, literalmente, 3.9% maior que o height da imagem, então apenas coloquei isso pra ele ir de forma dinâmica. Sobre trocar de 'rem' para 'px', prefiro deixar em 'rem'.

Sobre evitar usar px, realmente, evito ao máximo em alguns casos! Obrigado pelo feedback.

Reply:
Obrigado por considerar meu comentário, Jan.

### 1.

Constantes são valores constantes. Em `font-size: 14px;`, `14px` é uma constante, um **valor** constante, que é diferente de uma variável constante.

### 2.

> Se diz sobre os '@media', eles são para estilizar telas com proporções diferentes

Estou ciente 👌

### 3.

> caso tenha alguma dica ou ideia de como refatorar, é bem vindo

Proponho um `mixin` com body, mas isso é para um refactor maior no futuro. Esse `mixin` se pareceria com isso:

```scss
@mixin mobile-version {
  $max: 920px;
  @media (max-width: $max) {
    @content($max);
  }
}
```

Seria usado assim:

```scss
@include mobile-version {
  // mobile code
}
```

Ou

```scss
@include mobile-version using ($max) {
  // mobile code
  // $max está disponível aqui dentro
}
```

Mas isso é puramente uma questão de leitura, claro. Menos caracteres para se digitar.

### 4.

> Os valores são: calc(o tamanho do height + a % que é maior no width).

> Coloquei a porcentagem definida especificamente, pois a width da imagem é, literalmente, 3.9% maior que o height da imagem, então apenas coloquei isso pra ele ir de forma dinâmica.

Esse tipo de informação deveria estar documentada no código ou com comentários ou com variáveis (recomendado).

### 5.

> Sobre trocar de 'rem' para 'px', prefiro deixar em 'rem'.

Esse não é o meu ponto.

Minha preferência pessoal para fontes é `pt` porque costumava editar coisas que deveriam ser visualizadas em PDF bem antes de começar a programar. Apesar de `pt` ser uma [medida absoluta](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#numbers_lengths_and_percentages) (seu valor é exato em todos os computadores e navegadores), as pessoas aprendem a usar `em` e `rem` como base e com o tempo isso se torna uma convenção. Todos são supostos a saber o que é `rem` e tê-la usado extensivamente ao longo do tempo.

Muita coisa poderia ser dinâmica, mas eu não vejo um caso onde `14px` é mais legível que `0.8rem` se tratando de fontes. Principalmente porque a diferença entre adicionar ou diminuir `1px` da medida de uma fonte é grande demais quando o ajuste é muito fino. Pixels com casas decimais são razoáveis em tão poucos casos (`letter-spacing` é um deles) que é sempre o ultimo recurso possível.

E por ultimo: `rem` é especialmente útil porque é relativo ao valor da propriedade `font-size` do elemento `html` (ou `:root`). Se seu objetivo com `rem` é representar um valor grande com números menores faria **sentido** fazer algo assim:

```scss
:root {
  font-size: 10px; // ou 20px
}
```

Onde cada `Nrem` é o mesmo que `calc(10px * N)` ou `calc(20px * N)`.

Meu ponto no fim das contas é: certas coisas podem ser ignoradas em função de consistência e facilidade.

Calcular valores em imagens é um caso onde não há muito o que fazer quanto à consistência de valores, mas se você vai mesmo ter de medir os pixels, porque não fazer isso literalmente e evitar complexidade de código e carga no navegador? Falo de ir em um programa de imagens e usar uma ferramenta para contar os pixels desde um dos lados até onde você quer. Eu posso fazer isso pra você se quiser. E com isso, ao envés de `calc(9rem + 3.9%)` você teria, digamos, `$img-alpha-offset-bottom: 341px`. Qual dessas é mais legível e menos sujeito a erros causados por valores que não são citados no código? Lembre que diversas extensões de navegador alteram o CSS arbitrariamente.

Se o seu valor é do tipo que jamais poderia ser adivinhado ou modificado então é melhor mesmo usar uma constante com valores absolutos, mas se for fazer isso, tente fazer com que quem não escreveu o código saiba do que se trata sem precisar inspecionar o DOM sempre que esquecer.

Esse pensamento pode ser extendido à como você escreve suas constantes: `31.25rem` é um número menor, mas `500px` é mais fácil de se lembrar e realizar lógica na sua cabeça na hora de modificar alguma coisa. Nem tanto sobre a unidade de medida, mais sobre o número 500 ser bem mais fácil de se trabalhar do que o número 31.25, e por aí vai. De novo: se precisar usar um valor difícil, use uma variável.

Reply:
@marcosdly RT